### PR TITLE
Figure.meca: Improve documentation

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

Some improvements of the documetation of `Figure.meca`:
 - [ ] rewrapp line length
 - [ ] add missing articels
 - [ ] shorten formulations

**Preview**: https://pygmt-dev--3517.org.readthedocs.build/en/3517/api/generated/pygmt.Figure.meca.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

**Comment by seisman** [comment]
@yvonnefroehlich What's the status of this PR? Do you think we can finish it in v0.14.0?

File: doc/conf.py [hunk 1]
```diff
@@ -206,7 +206,7 @@
 
 # Configure theme
 html_theme = "sphinx_rtd_theme"
-html_theme_options = {}
+html_theme_options = {"version_selector": True}
```
Comment (yvonnefroehlich) [inline doc/conf.py:209]:
```suggestion
html_theme_options = {}
```

**Comment by yvonnefroehlich** [comment]
> @yvonnefroehlich What's the status of this PR? Do you think we can finish it in v0.14.0?

I just resolved the merge conflicts and will look later what is missing here.

File: pygmt/src/meca.py [hunk 2]
```diff
@@ -305,91 +300,84 @@ def meca(  # noqa: PLR0912, PLR0913, PLR0915
...
+        Latitude(s) of the event location(s). Length must match the number of events.
+        Overrides the ``latitude`` values in ``spec`` if ``spec`` is a dictionary or
+        :class:`pandas.DataFrame`.
     depth : float, list, or 1-D numpy array
-        Depth(s) of event location(s) in kilometers. Must be the same length
-        as the number of events. Will override the ``depth`` values in ``spec``
-        if ``spec`` is a dictionary or :class:`pandas.DataFrame`.
+        Depth(s) of the event location(s) in kilometers. Length must match the number
+        of events. Overrides the ``depth`` values in ``spec`` if  ``spec`` is a
+        dictionary or :class:`pandas.DataFrame`.
```
Comment (seisman) [inline pygmt/src/meca.py:317]: Combine docstrings of the three parameters into one entry? E.g.,
```
    longitude/latitude/depth : float, list, or 1-D numpy array
        Longitude(s)/latitude(s)/depth(s) of the events. Length must match the number
        of events. Overrides the ``longitude``/``latitude``/``depth`` values in 
        ``spec`` if  ``spec`` is a dictionary or :class:`pandas.DataFrame`.
```

Comment (yvonnefroehlich) [reply pygmt/src/meca.py]: Combined the docstring for `longitude`, `latitude`, and `depth` as well as for `plot_longitude` and `plot_latitude` into one entry, respectively.

File: pygmt/src/meca.py [hunk 3]
```diff
@@ -305,91 +300,72 @@ def meca(  # noqa: PLR0912, PLR0913, PLR0915
...
         - ``"full"``: the full seismic moment tensor
-        - ``"dc"``: the closest double couple defined from the moment tensor
-          (zero trace and zero determinant)
+        - ``"dc"``: the closest double couple defined from the moment tensor (zero
+          trace and zero determinant)
         - ``"deviatoric"``: deviatoric part of the moment tensor (zero trace)
-    longitude : float, list, or 1-D numpy array
-        Longitude(s) of event location(s). Must be the same length as the
-        number of events. Will override the ``longitude`` values
+    longitude / latitude / depth : float, list, or 1-D numpy array
```
Comment (seisman) [inline pygmt/src/meca.py:306]: With whitespaces, the doc doesn't look correct.
```suggestion
    longitude/latitude/depth : float, list, or 1-D numpy array
```

File: pygmt/src/meca.py [hunk 4]
```diff
@@ -305,91 +300,72 @@ def meca(  # noqa: PLR0912, PLR0913, PLR0915
...
-        as the number of events. Will override the ``depth`` values in ``spec``
-        if ``spec`` is a dictionary or :class:`pandas.DataFrame`.
-    plot_longitude : float, str, list, or 1-D numpy array
-        Longitude(s) at which to place beachball(s). Must be the same length
-        as the number of events. Will override the ``plot_longitude`` values
-        in ``spec`` if ``spec`` is a dictionary or :class:`pandas.DataFrame`.
-    plot_latitude : float, str, list, or 1-D numpy array
-        Latitude(s) at which to place beachball(s). List must be the same
-        length as the number of events. Will override the ``plot_latitude``
+    plot_longitude / plot_latitude : float, str, list, or 1-D numpy array
```
Comment (seisman) [inline pygmt/src/meca.py:310]:
```suggestion
    plot_longitude/plot_latitude : float, str, list, or 1-D numpy array
```

File: pygmt/src/meca.py [hunk 5]
```diff
@@ -227,74 +227,69 @@ def meca(  # noqa: PLR0912, PLR0913, PLR0915
...
+            beachball. ``0 0`` plots the beachball at the longitude and latitude
+            given in the columns 1 and 2. [optional; requires ``offset=True``].
+          - Last Column: text string to appear near the beachball [optional].
+
+        - *1-D np.array*: focal mechanism parameters of a single event.
           The meanings of columns are the same as above.
-        - *2-D array*: focal mechanism parameters of multiple events.
+        - *2-D np.array*: focal mechanism parameters of multiple events.
           The meanings of columns are the same as above.
         - *dictionary or :class:`pandas.DataFrame`*: The dictionary keys or
```
Comment (seisman) [inline pygmt/src/meca.py:251]:
```suggestion
        - *dict* or :class:`pandas.DataFrame`: The dictionary keys or
```

File: pygmt/src/meca.py [hunk 6]
```diff
@@ -227,74 +227,69 @@ def meca(  # noqa: PLR0912, PLR0913, PLR0915
...
+          convention. For the different conventions, the following combination of keys/
+          column names are required:
+
+          - ``"aki"``: *strike*, *dip*, *rake*, *magnitude*
+          - ``"gcmt"``: *strike1*, *dip1*, *rake1*, *strike2*, *dip2*, *rake2*,
+            *mantissa*, *exponent*
+          - ``"mt"``: *mrr*, *mtt*, *mff*, *mrt*, *mrf*, *mtf*, *exponent*
+          - ``"partial"``: *strike1*, *dip1*, *strike2*, *fault_type*, *magnitude*
+          - ``"principal_axis"``: *t_value*, *t_azimuth*, *t_plunge*, *n_value*,
+            *n_azimuth*, *n_plunge*, *p_value*, *p_azimuth*, *p_plunge*, *exponent*
```
[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply pygmt/src/meca.py]: I think this is a good idea. A table makes these different options more readable. And later we can point to this table in the beachball tutorial and short the introduction part there.

**Review by seisman — APPROVED** [review]

**Pull request merged**